test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/Extensibility/NuGetProjectServiceTests.cs

```diff
@@ -189,6 +191,21 @@ protected override Task<string> GetAssetsFilePathAsync(bool shouldThrow)
             {
                 throw new NotImplementedException();
             }
+
+            protected override Task<PackageSpec> GetPackageSpecAsync(CancellationToken ct)
+            {
+                throw new NotImplementedException();
+            }
+
+            protected override IEnumerable<PackageReference> FetchInstalledPackagesList(IEnumerable<LibraryDependency> libraries, NuGetFramework targetFramework, IList<PackageReference> installedPackages, IList<LockFileTarget> targets)
+            {
+                throw new NotImplementedException();
+            }
+
+            protected override IReadOnlyList<PackageReference> FetchTransitivePackagesList(NuGetFramework targetFramework, IList<PackageReference> installedPackages, IList<PackageReference> transitivePackages, IList<LockFileTarget> targets)
+            {
+                throw new NotImplementedException();
+            }
```

Comment:
Shouldn't these be `abstract` rather than `throw new NotImplementedException()`?